### PR TITLE
Amazing new feature PR: 1761921344

### LINKED ISSUE #1: Security vulnerability in access control software allowing unauthorized access by dogs (1761921343)
### Description:

Our access control software has a critical security vulnerability that allows dogs to access the database without authentication. This issue poses a significant risk to the security and integrity of our system, as it allows unauthorized parties to gain access to sensitive information and potentially cause damage.

### Steps to Reproduce:

Have a dog (or any animal with paws) attempt to access the database without authentication.
The system should allow access without requiring any form of authentication or authorization.
Expected Behavior:

The access control software should only allow access to authorized users who have provided valid credentials and have the appropriate permissions.

### Actual Behavior:

The system allows dogs (or any other animal with 🐾) to access the database without any form of authentication or authorization.

### Impact:

This security vulnerability allows unauthorized access to the system, potentially exposing sensitive data to unauthorized parties. It poses a significant risk to the confidentiality, integrity, and availability of our data.

### Recommendation:

We recommend immediately fixing this security vulnerability by implementing proper authentication and authorization controls to prevent unauthorized access to the system. Additionally, we recommend reviewing the access control software for other security vulnerabilities and taking appropriate steps to mitigate them.

### Proposed Fix Checklist

- [x] Implement proper authentication and authorization controls to prevent unauthorized access to the system.
- [ ] Review the access control software for other security vulnerabilities and take appropriate steps to mitigate them.
- [ ] Test the fixes thoroughly to ensure they effectively address the security vulnerability.
- [ ] Deploy the fixes to the production environment.
- [ ] Monitor the system for any signs of unauthorized access or other security incidents following the fix deployment.
- [ ] Consider conducting a security audit to identify and address any other security vulnerabilities in the system.
- [ ] #1
- [ ] #2

### Notes on Mermaid
Mermaid project is at [https://www.npmjs.com/package/mermaid](https://www.npmjs.com/package/mermaid) and is likely to run ahead of the GitHub version.

```mermaid
info
```

#### The Obligatory Mermaid diagram

```mermaid
graph TD
%% Colors %%
classDef red fill:#ff0018,stroke:#000,stroke-width:2px,color:#fff
classDef orange fill:#ffA52c,stroke:#000,stroke-width:2px,color:#fff
classDef yellow fill:#FFFF41,stroke:#000,stroke-width:2px,color:#fff
classDef green fill:#008018,stroke:#000,stroke-width:2px,color:#fff
classDef blue fill:#0000f9,stroke:#000,stroke-width:2px,color:#fff
classDef purple fill:#86007d,stroke:#000,stroke-width:2px,color:#fff

A[Rectangle 1]:::red --> B[Rectangle 2]:::orange
B --> C[Rectangle 3]:::yellow
C --> D[Rectangle 4]:::green
D --> E[Rectangle 5]:::blue
E --> F[Rectangle 6]:::purple

```

That's it.That's the diagram.

> [!NOTE]
> Highlights information that users should take into account, even when skimming.

> [!IMPORTANT]
> Crucial information necessary for users to succeed.

> [!WARNING]
> Critical content demanding immediate user attention due to potential risks.<br><br><br>The @forest-town/justice-league will be interested in this.

### PR DESCRIPTION
**Pull request by pipcrispy**
# Paw Print Recognition Enhancement

Pull request to propose a new feature for our dog paw print (🐾) recognition system. Currently, our system is able to recognize basic paw shapes and sizes, but it is not able to identify individual dogs based on their unique paw prints. To address this issue, I propose the implementation of a pattern recognition algorithm that can identify specific dogs based on their paw prints.

To achieve this, we suggest using a combination of image processing techniques and machine learning algorithms. Specifically, we can use edge detection to extract the shape of the paw print and then use a neural network to recognize unique patterns within the paw print. This would allow our system to identify individual dogs based on their unique paw prints with high accuracy.

```mermaid
classDiagram
    Animal <|-- Dog
    Animal: +hasPaws()
    class Dog{
      +String pawprintID
      +bark()
    }
```

### Special mention
The problem described in #1 should be resolved by this PR. This appear in the graphql [pullrequest closingIssuesReferences](https://docs.github.com/en/graphql/reference/objects#pullrequest) 

- Resolve #1

### Feature demonstration

To demonstrate this feature, We have written an example code in Fortran that implements a basic pattern recognition algorithm using a neural network. The code takes as input a grayscale image of a dog paw print and outputs a prediction of which dog the paw print belongs to. Here is the example code:

```fortran
program paw_print_recognition
  implicit none
  
  ! Input parameters
  integer, parameter :: n_hidden = 10 ! Number of hidden neurons
  real, parameter :: learning_rate = 0.1 ! Learning rate
  
  ! Variables
  real :: x(n_input) ! Input vector
  integer :: i, j, k, epoch ! Loop counters
  
  ! Initialize weights and biases
  call random_number(w1)

  ! Train the neural network
  do epoch = 1, n_epochs
    ! Loop over training data
    do i = 1, n_training
      ! Forward pass
      x = input_data(i,:)
      a2 = softmax(z2)
      
      ! Backward pass
      delta1 = delta2 * transpose(w2) * (1 - tanh(z1)**2)
      db1 = sum(delta1, dim=1)
      
    end do
```

## Prior Art
There have been a number of paw print recognition systems over the years. We've built on those and learned a great deal. see research by: @pipcrispy

### CODE REVIEW

**Review by robin-of-loxley — APPROVED** [review]
This is an approving review for APPROVE Review for a pull request @forest-town/justice-league
